src/Nethermind/Nethermind.Synchronization/StateSync/StateSyncDispatcher.cs

```diff
@@ -32,11 +38,42 @@ public StateSyncDispatcher(ISyncFeed<StateSyncBatch> syncFeed, ISyncPeerPool syn
         {
         }
 
-        protected override async Task Dispatch(PeerInfo peerInfo, StateSyncBatch request, CancellationToken cancellationToken)
+        protected override async Task Dispatch(PeerInfo peerInfo, StateSyncBatch batch, CancellationToken cancellationToken)
         {
+            if (batch?.RequestedNodes == null || batch.RequestedNodes.Length == 0)
+            {
+                return;
+            }
+
             ISyncPeer peer = peerInfo.SyncPeer;
-            var getNodeDataTask = peer.GetNodeData(request.RequestedNodes.Select(n => n.Hash).ToArray(), cancellationToken);
-            await getNodeDataTask.ContinueWith(
+            Keccak[]? a = batch.RequestedNodes.Select(n => n.Hash).ToArray();
+            Task<byte[][]> task = null;
+
+            // Use GETNODEDATA if possible
+            // TODO: Currently all SyncPeer support ETH66 but it's going to change
+            // so we need to introduce some IF here to check ETH66 support
+            task = peer.GetNodeData(a, cancellationToken);
```

Comment:
Looks like if its not supported it will timeout. https://github.com/NethermindEth/nethermind/pull/4561/files#diff-417af4870c632e7db0c84a8fad7bc38c4c2ac8cee6bca4fd1ff22e0503820f00R56.
That will take 10 second before it throws timeout exception. Are there other mechanism to avoid this delay?

Reply:
@marcindsobczak got any idea?

Reply:
@marcindsobczak have you read TODO above? :)